supercommand.go

```diff
@@ -524,12 +524,10 @@ func (c *SuperCommand) Run(ctx *Context) error {
 	if err != nil && !IsErrSilent(err) {
 		// Handle formatting when displaying errors.
 		handleErr := c.handleErrorForMachineFormats(ctx)
-		if handleErr != nil {
+		if handleErr != nil && handleErr != ErrSilent {
```

Comment:
This change will effectively suppress silent errors while the previous implementation would return the error even if it was ErrSilent.  Is this the intended behavior?